weld/syntax/parser.rs

```diff
@@ -1212,6 +1247,13 @@ impl<'t> Parser<'t> {
         let mut annotations = Annotations::new();
         try!(self.parse_annotations(&mut annotations));
 
+        if !self.peek().signals_type() {
+            let name = self.symbol()?.to_string();
+            // NOTE: The type field is not used right now, but may be later if we decide
+            // to keep type alias information past macro substitution.
+            return Ok(Alias(name, Box::new(Unknown)));
```

Comment:
One question here: it seems that only the last redeclaration of the same name should be preserved in the AST, otherwise there might be issues resolving the type (depending on whether ordering information is preserved / considered during type inference)?

Reply:
Each name should have a single type by the type the expression is processed (namely, the last definition for the name), so there shouldn't be conflicts. The definitions themselves won't ever have aliases since they're processed iteratively. E.g., if you have:

```
type int = i32;
type pair = {int,int};
type pairpair = {pair,pair};
```

When `pairpair` is processed, `pair` is already processed to be `{i32,i32}`.